src/main/java/io/cryostat/agent/remote/EventTemplatesContext.java

```diff
@@ -43,7 +43,7 @@ class EventTemplatesContext implements RemoteContext {
 
     @Override
     public String path() {
-        return "/event-templates";
+        return "/event-templates/";
```

Comment:
Somehow this is causing the HTTP connection to close, which triggers an exception when trying to create a recording for the agent

Reply:
Only "ALL" event template is available when trying to create a recording

Reply:
I believe the second part about only "ALL" being available happens because of how the JDK HTTP server handles request path resolution. `/event-templates` will match any path prefixed with that string (ex. it would match `/event-templates` but also `/event-templates-and-more` or `/event-templates/and/more`), whereas `/event-templates/` will require the request to be exactly `/event-templates/`.

The Cryostat server would previously send a request on `/event-templates`, but in this PR:

https://github.com/cryostatio/cryostat/pull/1604/files#diff-2241c6f25048112e06962b86cb0f144af726370b545c86c3e9b0e49e0fe29bb5R272

I updated it so that the server requests `/event-templates/`.

If there is a mismatch where the Agent API exposes/expects `/event-templates/` and the Cryostat server asks it for `/event-templates`, I think the Agent responds to the server with an HTTP 404, which the server handles by logging an exception and using an empty list as the response for the available remote target templates. That empty list gets merged with any custom templates on the server, plus the `"ALL"` meta-template.

Reply:
oops sorry I'm silly and didn't see the corresponding cryostat PR